source/extensions/original_ip_detection/xff/config.cc

```diff
@@ -0,0 +1,26 @@
+#include "extensions/original_ip_detection/xff/config.h"
+
+#include "envoy/extensions/original_ip_detection/xff/v3/xff.pb.h"
+#include "envoy/extensions/original_ip_detection/xff/v3/xff.pb.validate.h"
+#include "envoy/http/original_ip_detection.h"
+#include "envoy/registry/registry.h"
+
+#include "extensions/original_ip_detection/xff/xff.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace OriginalIPDetection {
+namespace Xff {
+
+Http::OriginalIPDetectionSharedPtr
+XffIPDetectionFactory::createExtension(const Protobuf::Message& message) const {
+  return std::make_shared<XffIPDetection>(
+      dynamic_cast<const envoy::extensions::original_ip_detection::xff::v3::XffConfig&>(message));
```

Comment:
MessageUtil::downcastAndValidate<> ?

Reply:
is that needed given from the HCM we do this before calling `createExtension()`:

```
      ProtobufTypes::MessagePtr message = factory->createEmptyConfigProto();
      Envoy::Config::Utility::translateOpaqueConfig(typed_config.typed_config(),
                                                    ProtobufWkt::Struct::default_instance(),
                                                    context_.messageValidationVisitor(), *message);
      original_ip_detection_extensions_.push_back(factory->createExtension(*message));
```

?

Reply:
Hm, all things being equal I'm more inclined to ust have the message stay a generic message in the HCM and do the validation here, as it's easier to have unit tests that your validation works (and we avoid a dyanmic cast).  WDYT?

Reply:
Yeah sounds more consistent, let me move in that direction.